runtime/domain.c

```diff
@@ -760,7 +784,21 @@ CAMLprim value caml_domain_spawn(value callback, value mutex)
   }
   init_domain_ml_values(p.ml_values, callback, mutex);
 
+#ifdef POSIX_SIGNALS
+/* we block all signals while we spawn the new domain, this is because
+   pthread_create inherits the current signals set and we want to avoid
+   a signal handler being triggered in the new domain before
+   the domain_state is fully populated. */
+  sigfillset(&mask);
+  pthread_sigmask(SIG_BLOCK, &mask, &old_mask);
+  p.mask = old_mask;
+#endif
   err = pthread_create(&th, 0, domain_thread_func, (void*)&p);
```

Comment:
It's sad to see so many  `#ifdef POSIX_SIGNALS`.  If you have POSIX threads, you have POSIX signals.  Since you're calling `pthread_create` directly, you have POSIX threads...

The only platform that does not have POSIX_SIGNALS is Win32.  On the one hand it's good to make provisions for a future Win32 port.  On the other hand a lot of the current code assumes POSIX threads and doesn't work with Win32.

Reply:
Yes, this is a good point. So much of `domain.c` will have to change for Win32 that it's better to just assume POSIX_SIGNALS. I have removed the ifdefs.